src/index.js

```diff
@@ -0,0 +1,175 @@
+
+          newData.forEach((item, indexItem) => {
+            item.index = indexItem;
+          });
+
+          localStorage.setItem('data', JSON.stringify(newData));
+          this.read();
+        }
+      });
+    });
+  }
+
+  getStoredData() {
+    const newData = JSON.parse(localStorage.getItem('data') || '[]');
+    this.newData = newData;
+    return newData;
+  }
+
+  Clear() {
+    clearSelected.addEventListener('click', () => {
+      const newdata = this.getStoredData().filter((data) => !data.completed);
+
+      newdata.forEach((item, index) => {
+        item.index = index;
+      });
+
+      localStorage.setItem('data', JSON.stringify(newdata));
+      // window.location.reload();
+      this.read();
+    });
+  }
+
+  update() {
+    const btnitems = document.querySelectorAll('#dot');
+    const Input = document.createElement('input');
+    const checkboxes = document.querySelectorAll('#checkbox');
+    Input.className = 'InputUpdate';
+    const newdata = this.getStoredData();
+
+    checkboxes.forEach((checkboxe) => {
+      checkboxe.addEventListener('change', (e) => {
+        this.getStoredData().forEach((item) => {
+          if (item.index === +e.target.dataset.id) {
+            item.completed = true;
+            const index = newdata.findIndex((item) => item.index === +e.target.dataset.id);
+            newdata[index].completed = true;
+
+            localStorage.setItem('data', JSON.stringify(newdata));
+          }
+        });
+        e.target.nextElementSibling.style.textDecoration = '2px black line-through';
+      });
+    });
+
+    btnitems.forEach((btn) => {
+      btn.addEventListener('click', (e) => {
+        const btntdot = e.target.parentElement;
+        const deletebtn = btntdot.nextElementSibling;
+        const formTarget = btntdot.previousElementSibling.children[2];
+        const h2 = formTarget.previousElementSibling;
+
+        btntdot.classList.add('hidden');
+        deletebtn.classList.remove('hidden');
+        deletebtn.style.color = 'red';
+        formTarget.style.display = 'flex';
+        h2.classList.add('hidden');
+
+        formTarget.addEventListener('submit', (e) => {
+          e.preventDefault();
+          const input = e.target.querySelector('input');
+          const { value } = input;
+          const itemIndex = input.dataset.parentindex;
+
+          const newget = [...this.getStoredData()];
+          formTarget.style.display = 'none';
+          btntdot.classList.remove('hidden');
+          deletebtn.classList.add('hidden');
+          h2.classList.remove('hidden');
+
+          newget.filter((data) => {
+            if (data.index === +itemIndex) {
+              const item = newget[itemIndex];
+              item.description = value;
+              localStorage.setItem('data', JSON.stringify(newget));
+              this.read();
+            }
+            return newget;
+          });
+        });
+      });
+    });
+  }
+
+  read() {
```

Comment:
I think it would be a lot better if you could use an arrow function here to make your codes look more updated :ok_hand: , what do you say about that  :thinking:  ?

Reply:
I think that this is a method class it can not be converted into an arrow function.